_posts/2021-12-27-trino-2021-a-year-of-growth.md

```diff
@@ -0,0 +1,346 @@
+precision larger than 19.
+* Improved performance of queries that process row or array data.
+* Reduced memory usage for processing `JOIN` clauses.
+* Improved performance of queries that contain a `DISTINCT` clause. 
+
+## Security
+
+* Added support for automatic configuration of TLS for 
+[Secure internal communication](https://trino.io/docs/current/security/internal-communication.html).
+This removes the need to provision per-worker TLS certificates.
+* Added support to use internal names for discovery client when automatic TLS is
+enabled for internal communications. This allows discovery.uri to be configured 
+using a normal DNS name like `https://coordinator.trino` and still use automatic
+TLS certificates.
+* Added support for OAuth2/OIDC opaque access tokens.
+* Added support for configuring multiple password authentication plugins.
+* Added property to hide inaccessible columns from `SELECT *` statement when the 
+`hide-inaccessible-columns` configuration property is set to true. 
+* Allowed configuring HTTP proxy for OAuth2 authentication.
+* Added `IN catalog` clause to `CREATE ROLE`, `DROP ROLE`, `GRANT ROLE`, 
+`REVOKE ROLE`, and `SET ROLE` to specify the target catalog of the statement 
+instead of using the current session catalog.
+* Fix handling of SNI for multiple TLS certificates.
+
+## Data Sources
+
+### BigQuery Connector
+
+* Added support for case insensitive name matching for BigQuery views.
+* Changed type mapping of BigQuery datetime from `timestamp(3)` to `timestamp(6)`
+in Trino.
+* Changed type mapping of BigQuery timestamp from `timestamp(3)` with time zone 
+to `timestamp(6)` with time zone in Trino.
+* Fixed failures for queries accessing `information_schema.columns` when 
+`case-insensitive-name-matching` is disabled.
+
+### Cassandra connector
+
+* Mapped Cassandra `uuid` type to Trino `uuid`.
+* Added support for Cassandra `tuple` type.
+* Changed minimum number of speculative executions from 2 to 1. 
+
+### Clickhouse connector (new)
+
+* Added [ClickHouse connector](https://trino.io/docs/current/connector/clickhouse.html).
+* Added `write.batch-size` connector configuration property to control JDBC batch 
+size used during writes. It can also be controlled using the `write_batch_size` 
+session property.
+* Added new configuration property `insert.non-transactional-insert.enabled` to 
+allow `INSERT` queries to write directly to the target table. This can improve 
+performance in some cases by sacrificing transactional behaviour. It can also be
+controlled using `non_transactional_insert` session property.
+* Improved performance of aggregation queries by computing aggregations within 
+ClickHouse. Currently, the following aggregate functions are eligible for
+pushdown: `count`, `min`, `max`, `sum` and `avg`.
+* Added support for dropping columns.
+* Map ClickHouse `UUID` columns as `UUID` type in Trino instead of `VARCHAR`.
+
+### HDFS, S3, Azure and cloud object storage systems
```

Comment:
Is this Hive connector? Or Iceberg connector? Or both?

Reply:
Both, this is how we had it last year.